server/test/services/openApi/v2/serializers/DefinitionSerializerTest.java

```diff
@@ -0,0 +1,40 @@
+package services.openApi.v2.serializers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.Test;
+import services.openApi.v2.Definition;
+import services.openApi.v2.DefinitionType;
+import services.openApi.v2.Format;
+
+public class DefinitionSerializerTest extends OpenApiSerializationAsserter {
+  @Test
+  public void canSerializeFullObject() throws JsonProcessingException {
+    Definition model =
+        Definition.builder("name1", DefinitionType.INTEGER)
+            .setFormat(Format.INT32)
+            .addDefinition(Definition.builder("name2", DefinitionType.STRING).build())
+            .build();
+
+    String expected =
+        new YamlFormatter()
+            .appendLine("name1:")
+            .appendLine("  type: integer")
+            .appendLine("  format: int32")
+            .appendLine("  properties:")
```

Comment:
Question: If something is of `type: integer`, can it also have `properties`?

Even if not, maybe we don't care about preventing it in this library, as it'll be caught by a schema validator.